### ISSUE TITLE
Force Plot Is not Displayed

### ISSUE THREAD

**Issue by Helmold** [shap/shap#1358]
Hey there I'm totally new into programming with python. Doing it for my thesis. I managed to learn a lot by reading all your issues but right now I'm a little desperate. 

The following: 

x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 0.3 , random_state=42)

dtrain = lgb.Dataset(x_train, y_train, free_raw_data=False, silent=True)
lgb_params = {
        'objective': 'binary',
        'boosting_type': 'rf',
        'subsample': 0.623,
        'colsample_bytree': 0.7,
        'num_leaves': 127,
        'max_depth': 8,
        'seed': None,
        'bagging_freq': 1,
        'n_jobs': 4
    }

explainer = shap.TreeExplainer(clf)
shap_values = explainer.shap_values(x_train)
print(shap_values)
shap.summary_plot(shap_values, x_train) 
shap.summary_plot(shap_values[1], x_train)
shap.force_plot(explainer.expected_value [1],shap_values[1][1,:],x_train.iloc[1,:])
shap.force_plot(explainer.expected_value[1],shap_values[1][:1000,:],x_train.iloc[:1000,:])

I get beautiful outputs for the summary plots. But the forceplot doesn't show up. If i add plt.savefigure and plt.show a blank plot is shown and saved. No error is reported. The program just finishes. I tried mathplotlib = true and it worked partly for the single force plot. 
How can i get the plot shown?
I tried shap.inijs , plt.show, display etc pp nothing worked

Thanks for helping

**Comment by XJrandom** [comment]
you can add plt.switch_backend('agg') at first in force_plot， and  add plt.savefig('summary_plot_result.jpg') besides if show

**Comment by Helmold** [comment]
Thanks I'll try this

**Comment by wmichalak** [comment]
I have a similar issue. 
```shap.force_plot(explainer.expected_value, shap_values[0, :], X_sample.iloc[0, :], matplotlib=True```
will show up in my interactive environment and I can see the results, but 
``` plt.savefig('force_plot.png')```
results in a blank/white figure.
Thus, the figure is useless if created by a script.

**Comment by chetanambi** [comment]
I also had similar issue. This below code working for me. I only added show=False.
```
shap.force_plot(explainer.expected_value, shap_values[0, :], X_sample.iloc[0, :], matplotlib=True, show=False)
plt.savefig('force_plot.png')
```

**Comment by tytcc** [comment]
I had a similar issue.
`shap.force_plot(explainer.expected_value, shap_values[0, :], X_sample.iloc[0, :], matplotlib=True)`
works for me. I added matplotlib=True.

**Comment by JTFouquier** [comment]
I'm also having the same issue and I'd really love to get this plot to work. The only way I could see any sort of force_plot was to add `matplotlib=True` _and_ `show=False` but it's looking very strange. It shows up in my Mac "preview" window and it's smooshed with the previous graphic, but when I open the whole PDF it's not there at all. Pics if they might help. Thanks!!

![Screen Shot 2021-07-25 at 4 32 50 PM](https://user-images.githubusercontent.com/7920357/126915473-a9811e56-ff52-4b24-92f5-a4d7acef4719.png)
![Screen Shot 2021-07-25 at 4 32 55 PM](https://user-images.githubusercontent.com/7920357/126915475-70d16b45-50cd-41da-b87c-48503b3a6812.png)
![Screen Shot 2021-07-25 at 4 33 28 PM](https://user-images.githubusercontent.com/7920357/126915476-096fc3c5-2706-421f-9a6f-abb1d58162db.png)

**Comment by Wilsonkongcheng** [comment]
@JTFouquier
> I'm also having the same issue and I'd really love to get this plot to work. The only way I could see any sort of force_plot was to add `matplotlib=True` _and_ `show=False` but it's looking very strange. It shows up in my Mac "preview" window and it's smooshed with the previous graphic, but when I open the whole PDF it's not there at all. Pics if they might help. Thanks!!
> 
> ![Screen Shot 2021-07-25 at 4 32 50 PM](https://user-images.githubusercontent.com/7920357/126915473-a9811e56-ff52-4b24-92f5-a4d7acef4719.png) ![Screen Shot 2021-07-25 at 4 32 55 PM](https://user-images.githubusercontent.com/7920357/126915475-70d16b45-50cd-41da-b87c-48503b3a6812.png) ![Screen Shot 2021-07-25 at 4 33 28 PM](https://user-images.githubusercontent.com/7920357/126915476-096fc3c5-2706-421f-9a6f-abb1d58162db.png)

Try it:
`shap.force_plot(explainer.expected_value[1], shap_values[1][i, :], data[cols].iloc[i, :], matplotlib=True,
                    show=True)
 plt.savefig(f"sample.jpg", bbox_inches='tight')`

**Comment by sunqingmeteo** [comment]
> I also had similar issue. This below code working for me. I only added show=False.
> 
> ```
> shap.force_plot(explainer.expected_value, shap_values[0, :], X_sample.iloc[0, :], matplotlib=True, show=False)
> plt.savefig('force_plot.png')
> ```

"show=False" works great for me.

**Comment by PatrickKlingler** [comment]
I am running 0.49.1 and when I run
```
shap.plots.force(explainer.expected_value, shap_explanation, X.loc[idx], link='logit', matplotlib=True, show=False)
plt.title(
    ...
)
plt.savefig(
    path,
    dpi=199,
    bbox_inches="tight",
)
```
it saves an empty figure with my title, tried a bunch of fussing about with the axes and facecolor but no dice. It works when I remove link='logit'






As context, this is just a workaround so I can output a waterfall-esque plot with probabilities using categorical features.

If there's a way to output probabilities without `feature_perturbation='interventional'` for the waterfall plot then that'd be great, though I saw there's an issue with the axes being non-linear or something.
